### ISSUE TITLE
[feature]: disallowing default exports should also fail for a dynamic import that assumes default exports

### ISSUE THREAD

**Issue by Aghassi** [benmosher/eslint-plugin-import#2029]
When users use `export default`, they can safely do `import(module).then( module => ... )` to use the module. If `export default` is disallowed users will hit a runtime error that there is no property `default` on the object returned from the promise. It feels like there should be a rule that pairs with this that validates dynamic imports of modules that don't provide a default export should be shaped like

```js
                  import('path/to/module')
                    .then(module) => ({
                      default: module
                    })
```

This way we teach folks proactively how to cope in a situation that would otherwise fail. Thoughts on this idea?

**Comment by ljharb** [comment]
No, they won't hit such a runtime error. `import()` resolves to a Promise for the module namespace object, not the default export.

**Comment by ljharb** [comment]
In other words, `module` is ALWAYS an object, which may or may not have a `default` property.

**Comment by Aghassi** [comment]
Ah you are right, I am wrong. Thanks for clarifying. I still feel like this area could be improved, but I don't have any ideas off hand on the best way to handle it. I will close this for now, but I appreciate the prompt response!
